blogpost/post.md

```diff
@@ -0,0 +1,252 @@
+replacement for the PyTorch API, as it is **much slower**.
+
+## Differences between NumPy and `torch.compile`d NumPy
+
+**NumPy scalars**. NumPy returns NumPy scalars in almost any case where PyTorch
+would return a 0-D tensor (e.g. from `np.sum`). Under `torch.compile`, NumPy
+scalars are treated as 0-D arrays. This is just fine in most cases. The only
+case when their behavior diverges is when NumPy scalars are implicitly used as
+Python scalars. For example,
+```python
+>>> np.asarray(2) * [1, 2, 3]  # 0-D array is an array-like
+array([2, 4, 6])
+>>> u = np.int32(2)
+>>> u * [1, 2, 3]              # scalar decays into a Python int
+[1, 2, 3, 1, 2, 3]
+>>> torch.compile(lambda: u * [1, 2, 3])()
+array([2, 4, 6])               # acts as a 0-D array, not as a scalar ?!?!
+```
+
+If we compile the first two lines, we see that `torch.compile` treats `u` as a
+0-D array. To recover the eager semantics, we just need to make the casting
+explicit
+```python
+>>> torch.compile(lambda: int(u) * [1, 2, 3])()
+[1, 2, 3, 1, 2, 3]
+```
+
+**Type promotion and versioning**. NumPy's type promotion rules may be, at
+times, a bit surprising
+```python
+>>> np.asarray([1], dtype=np.int8) + 127
+array([128], dtype=int8)
+>>> np.asarray([1], dtype=np.int8) + 128
+array([129], dtype=int16)
+```
+These rules are changing to follow a set of rules that is closer to that of
+PyTorch in NumPy 2.0. The relevant technical document is [NEP 50](https://numpy.org/neps/nep-0050-scalar-promotion.html).
+`torch.compile` went ahead and implemented NEP 50 rather than the about-to-be-deprecated rules.
+
+In general, `torch.compile` will match the semantics of the last NumPy release.
+
+## Beyond NumPy: SciPy and scikit-learn
+
+In parallel to this effort, other Quansight engineers have designed, proposed
+and got merged a way to support PyTorch arrays within SciPy and scikit-learn.
+This was encountered with a big enthusiasm by the other maintainers from these
+libraries, as it was shown that using PyTorch as a backend would often yield
+considerable speed-ups.
+
+This can of course be combined with `torch.compile`  to be able to compile
+programs that rely on these other libraries.
+
+Note that the initial support is just restricted to a few algorithms in
+scikit-learn and to `scipy.cluster` in SciPy.
+
+If you want to learn more about this effort, how to use it, or how to help
+moving it forward, see this post. [TODO link post]
+
+## Conclusion [TODO Make sure Greg approves this wording] PyTorch has committed
+since its inception to be a framework compatible with the rest of the Python
+ecosystem. Enabling compiling NumPy programs, and establishing the tools
+necessary to do the same for other prominent libraries are two more steps in
+this direction. Quansight and Meta continue working in this direction,
+improving the compatibility between PyTorch and the rest of the ecosystem.
+
+From Quansight, we would like to thank Meta for funding this project and all
```

Comment:
I suggest to change "and all the previous work ..." to "as well as previous work on improving NumPy compatibility within PyTorch".